snowfakery_samples/npsp/RD_npsp.recipe.yml

```diff
@@ -1,72 +1,132 @@
-- object: Account
-  nickname: A Company
-  fields:
-    Name:
-      fake: last_name
-    RecordType: Organization
+### ------------- [          ORDER OF INSERT         ]------------ ###
+# 1-0. Org Account (5)
+# 1-1. Contact (5)
+# 1-2. Define MACRO for npe03__Recurring_Donation__c
+# 1-3. npe03__Recurring_Donation__c (5) - Org Account
+# 1-4. npe03__Recurring_Donation__c (5) - Household
 
+### ----------- [ RD_npsp.recipe.yml Summary ]---------- ###
+# This recipe creates a set of 10 Recurring Donation records, at random # of Open-ended and Fixed-length donations, 
+# related to 5 Contact and 5 Organization Account.
+# This recipe is specifically for Enhanced Recurring Donations
 
-- object: Contact
-  fields:
-    FirstName:
-      fake: first_name
-    LastName:
-      fake: last_name
+### ----------- [               CLI RUN                ]---------- ###
+
+# cci task run generate_and_load_from_yaml --generator_yaml snowfakery_samples/npsp/RD_npsp.recipe.yml --num_records 300 --num_records_tablename npe03__Recurring_Donation__c --org qa
+# snowfakery --output-format json --output-file src/foo.json snowfakery_samples/npsp/RD_npsp.recipe.yml
+
+### ----------- [            Recipe Starts             ]---------- ###
 
 - macro: commomfields
   fields:
     npe03__Amount__c:
       random_number:
         min: 10
         max: 500
-    npsp__Day_of_Month__c:
-      random_number:
-        min: 1
-        max: 28
-    npsp__DisableFirstInstallment__c: true
-    npsp__StartDate__c: 
-      date_between:
-        start_date: 2020-01-01
-        end_date: today
-    npsp__InstallmentFrequency__c: 1
+
     npe03__Installment_Period__c: 
       random_choice:
         - Monthly
         - Yearly
         - Weekly
         - 1st and 15th
+    npsp__Day_of_Month__c:
+      if:
+        - choice:
+            when: ${{npe03__Installment_Period__c == 'Monthly'}}
+            pick: ${{random_number(min=1,max=29)}}
+        - choice:
+            when: ${{npe03__Installment_Period__c != 'Monthly'}}
+            pick: NULL
+
+    # uncomment this line if you want to disable the first installment of Opportunity Auto-Creation
+    #npsp__DisableFirstInstallment__c: true
+    npsp__StartDate__c: 
+      date_between:
+        start_date: -180d
+        end_date: +120d
+    npsp__InstallmentFrequency__c:
```

Comment:
And one last suggestion, @eehjunggnujhee . I promise. :) NPSP defaults the Date Established field (npe03__Date_Established__c) to today's date. But if we are generating all sorts of random dates for the Effective Date but the Date Established is today, it might also be a a bit off, data-wise. Is it possible to add **_one_** last line to set the Date Established to the date you set in your Effective Date (npsp__StartDate__c) field? 
`    npe03__Date_Established__c: ${{npsp__StartDate__c}}
`
Thank you!!! Really fantastic work.

Reply:
no problem, I was wondering about that, too. updated!

Reply:
Awesome. Looks great! Will merge now. Thanks again!